test/algorithms/distribution_learners/qgan/test_qgan.py

```diff
@@ -266,14 +266,18 @@ def test_qgan_training_run_algo_numpy_multivariate(self):
 
     def test_qgan_training_analytic_gradients(self):
         """Test QGAN training with analytic gradients"""
-        self.qgan.set_generator(self.generator_circuit)
-        numeric_results = self.qgan.run(self.qi_qasm)
-        self.qgan.set_generator(self.generator_circuit,
-                                generator_gradient=Gradient('param_shift'))
-        analytic_results = self.qgan.run(self.qi_qasm)
-        self.assertAlmostEqual(numeric_results['rel_entr'],
-                               analytic_results['rel_entr'], delta=0.1)
-
+        for backend in ['qasm', 'sv']:
```

Comment:
It would be preferable to use `ddt` here so the backend ends up being passed to test so we run two independent tests instead of one that will stop at the first failure in the loop .